src/list.c

```diff
@@ -1,17 +1,56 @@
 #include <stdlib.h>
+#include <string.h>
 
 #include "entry.h"
 #include "list-private.h"
 #include "list.h"
 
+/* Função que cria e inicializa uma nova lista (estrutura list_t a
+ * ser definida pelo grupo no ficheiro list-private.h).
+ * Retorna a lista ou NULL em caso de erro.
+ */
 struct list_t *list_create() {
     struct list_t *list = (struct list_t *) calloc(1, sizeof(struct list_t));
```

Comment:
por curiosidade, porquê calloc e não malloc

Reply:
The first reason was because `calloc` inits the contents to `0`. There's another reason to use `calloc` instead of `malloc` in some scenarios, which is it allows you to allocate `n` blocks of a fixed size. 

The reason here was only to init the mem block to all zeroes. It basically makes it so `list->size == 0` and `list-head == NULL`, from the get go.